src/main.cpp

```diff
@@ -1295,7 +1295,8 @@ bool ReadBlockFromDisk(CBlock& block, const CDiskBlockPos& pos)
     }
 
     // Check the header
-    if (!CheckProofOfWork(block.GetHash(), block.nBits, Params().GetConsensus()))
+    if (!(CheckEquihashSolution(&block, Params()) &&
+          CheckProofOfWork(block.GetHash(), block.nBits, Params().GetConsensus())))
```

Comment:
This appears to check that that (1) the Equishash solution in the block is valid, and (2) the hash of the block is less than the difficulty target (or whatever it's called). What's stopping me from iterating over the order of transactions, content of transactions, etc. in order to bring the block hash's value down without finding more Equihash solutions? One way is to make the equihash problem depend on a hash of all the other contents of the block except for the equihash solution. I suspect that's what was done but I haven't seen code that does that yet.